packages/features/eventtypes/components/tabs/assignment/EventTeamAssignmentTab.tsx

```diff
@@ -368,39 +476,74 @@ const RoundRobinHosts = ({
             )}
           />
         </>
-        <AddMembersWithSwitch
-          placeholder={t("add_a_member")}
-          teamId={teamId}
-          teamMembers={teamMembers}
-          value={value}
-          onChange={onChange}
-          assignAllTeamMembers={assignAllTeamMembers}
-          setAssignAllTeamMembers={setAssignAllTeamMembers}
-          isSegmentApplicable={isSegmentApplicable}
-          automaticAddAllEnabled={true}
-          isRRWeightsEnabled={isRRWeightsEnabled}
-          isFixed={false}
-          containerClassName={assignAllTeamMembers ? "-mt-4" : ""}
-          onActive={() => {
-            const currentHosts = getValues("hosts");
-            setValue(
-              "hosts",
-              teamMembers.map((teamMember) => {
-                const host = currentHosts.find((host) => host.userId === parseInt(teamMember.value, 10));
-                return {
-                  isFixed: false,
-                  userId: parseInt(teamMember.value, 10),
-                  priority: host?.priority ?? 2,
-                  weight: host?.weight ?? 100,
-                  // if host was already added, retain scheduleId
-                  scheduleId: host?.scheduleId || teamMember.defaultScheduleId,
-                };
-              }),
-              { shouldDirty: true }
-            );
-          }}
-          customClassNames={customClassNames?.addMembers}
-        />
+        {!hostGroups.length ? (
+          createAddMembersWithSwitch(hostGroups[0]?.id ?? null)
+        ) : (
+          <>
+            {/* Show unassigned hosts first */}
+            {(() => {
+              const unassignedHosts = value.filter((host) => !host.isFixed && !host.groupId);
+              if (unassignedHosts.length > 0) {
+                return (
+                  <div className="border-subtle my-4 rounded-md border p-4 pb-0">
+                    <div className="-mb-1 flex items-center justify-between">
+                      <div className="flex items-center gap-2">
+                        <span className="text-default text-sm font-medium">{`Group ${
+                          hostGroups.length + 1
+                        }`}</span>
+                      </div>
+                    </div>
+                    {createAddMembersWithSwitch(null)}
+                  </div>
+                );
+              }
+              return null;
+            })()}
+
+            {/* Show all defined groups */}
+            {hostGroups.map((group, index) => {
+              const groupNumber = index + 1;
+
+              return (
+                <div key={index} className="border-subtle my-4 rounded-md border p-4 pb-0">
+                  <div className="-mb-1 flex items-center justify-between">
+                    <div className="flex items-center gap-2">
+                      <input
+                        type="text"
+                        value={group.name ?? ""}
+                        onChange={(e) => {
```

Comment:
As Sean mentioned elsewhere, it'd nice to extract these handlers as functions before return clause, so that we have most of the logics up there. Plus, these handlers will have proper names, which is another win for us.